src/main/java/org/codehaus/plexus/archiver/AbstractUnArchiver.java

```diff
@@ -343,9 +343,14 @@ protected void extractFile( final File srcF, final File dir, final InputStream c
 
         try
         {
-            if ( !isOverwrite() && f.exists() && ( f.lastModified() >= entryDate.getTime() ) )
+            if ( f.exists() && !StringUtils.equalsIgnoreCase( entryName, canonicalDestPath ) )
             {
-                return;
+                String message = String.format( "Archive entry %s and existing file %s names differ only by case."
+                        + " This may cause issues on case insensitive file systems.", entryName, canonicalDestPath );
+                getLogger().warn( message );
+                if ( !isOverwrite() ) {
```

Comment:
I would expect `( f.lastModified() >= entryDate.getTime() )` here to stay.

    entryname  | entrydate | filename   | filedate | behavior
    readme.txt | 1970      | readme.txt | 2020     | never overwrite
    readme.txt | 2020      | readme.txt | 1970     | only overwrite when isOverWrite()
    README.txt | 1970      | readme.txt | 2020     | warn + never overwrite
    README.txt | 2020      | readme.txt | 1970     | warn + only overwrite when isOverWrite()

I can extend this table when required

Reply:
Done. Also added tests that verify the warning is logged.

Reply:
You mentioned #3 is incorrect: why warn if there is no override? Did you change that?

Reply:
No, I did not change that - I strictly followed the example table.

Reply:
@plamentotev and/or @michael-o, any thoughts on rule number 3? Would it make sense to log the warning when we're not going to overwrite anyway?

Reply:
I think it would make sense to warn the user. If the file system is case sensitive in case 3 we are going to have two files (README.txt and readme.txt) but on case insensitive file systems we are going to end with single file. For me it does not matter if readme.txt is overwritten or not - the result on case insensitive filesystems is not the expected one.

Reply:
I've considered logging the warning on all types of file systems in scenario 3, but I'm afraid that's not feasible:

Finding out whether there is any casing variant on the canonical target file name is very expensive.
It would require to list the directory contents and do an `equalsIgnoreCase` on every file we find there
comparing it to the canonical target file name. README.txt isn't there, but is there ReAdMe.TxT, or REadME.tXt, or ...?
I don't think that would be a good idea :-).

I agree that having readme.txt (existing file) and README.txt (from the archive) next to each other is a strange situation, but it's not something _we_ chose and it doesn't conflict either. I'd argue that in such situations a warning is not worth the effort.

Reply:
In the end it looks like we have 2 tastes: 
- make case-sensitive OS users aware the jar will overwrite a file on case-insensitive OS systems.
- warn case-insensitive OS users that is overwriting a file with a different name.

I can imagine that if develop teams are all using case-sensitive OS, the warning doesn't add value.
For that reason I agree that we could go for the second taste.
And if it is an issue, we could always add a parameter to enforce equal OS independent results, but I consider that issue too rare for now.

Reply:
@rfscholte How do you want to guarantee the second taste? There are a number of factors to determine case-sensitivity.

Reply:
AFAIK the JRE takes care of that: new File("readme.txt").exists() should return true even it is actually "README.txt"

Reply:
Yes, you are right, of course. Complete forgot that.

Reply:
> AFAIK the JRE takes care of that: new File("readme.txt").exists() should return true even it is actually "README.txt"

Double-checked that last Friday, can confirm for Windows, macOS and Linux.